Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -258,19 +321,42 @@ class SmartPage_Context
     }
   };
 
-  public SmartPage_PageRequestManager_endRequest = function (sender: unknown, args: Sys.WebForms.EndRequestEventArgs)
+  public OverrideErrorMessageForStatusCode (statusCode: number, message: string) : void {
+    let current = this._errorMessageMap.get(statusCode);
+    if (current == undefined) return;
+
+    current.errorMessage = message;
+  }
+
+  public SmartPage_PageRequestManager_endRequest = (sender: unknown, args: Sys.WebForms.EndRequestEventArgs) =>
   {
-    if (args.get_error() != undefined && (args.get_error() as any).httpStatusCode == '500')
+    if (args.get_error() != undefined)
     {
-      const errorMessage = args.get_error().message;
-      args.set_errorHandled(true);
+      let statusCode = (args.get_error() as any).httpStatusCode;
 
-      const errorBody = '<div class="SmartPageErrorBody"><div>' + errorMessage + '</div></div>';
+      let errorDeclaration = this._errorMessageMap.get(statusCode);
 
-      SmartPage_Context.Instance!.ShowMessage("SmartPageServerErrorMessage", errorBody);
+      if (errorDeclaration == undefined || statusCode == 200){
```

Comment:
Please always use the non-type coercion variants of `==` and `!=`. These are the triple variants `===` and `!==`. New code should always use the triple variants as there is no implicit type conversion before comparing the values.